src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs

```diff
@@ -14,23 +14,66 @@ public class Meter : IDisposable
     {
         private static readonly List<Meter> s_allMeters = new List<Meter>();
         private List<Instrument> _instruments = new List<Instrument>();
+        private Dictionary<string, List<Instrument>> _nonObservableInstrumentsCache = new();
         internal bool Disposed { get; private set; }
 
+        /// <summary>
+        /// Initialize a new instance of the Meter using the <see cref="MeterOptions" />.
+        /// </summary>
+        public Meter(MeterOptions options)
+        {
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            Debug.Assert(options.Name is not null);
+
+            Initialize(options.Name, options.Version, options.Tags, options.Scope);
+
+            Debug.Assert(Name is not null);
+        }
+
         /// <summary>
         /// Initializes a new instance of the Meter using the meter name.
         /// </summary>
         /// <param name="name">The Meter name.</param>
-        public Meter(string name) : this (name, null) {}
+        public Meter(string name) : this (name, null, null, null) {}
 
         /// <summary>
         /// Initializes a new instance of the Meter using the meter name and version.
         /// </summary>
         /// <param name="name">The Meter name.</param>
         /// <param name="version">The optional Meter version.</param>
-        public Meter(string name, string? version)
+        public Meter(string name, string? version) : this (name, version, null, null) {}
+
+        /// <summary>
+        /// Initializes a new instance of the Meter using the meter name and version.
+        /// </summary>
+        /// <param name="name">The Meter name.</param>
+        /// <param name="version">The optional Meter version.</param>
+        /// <param name="tags">The optional Meter tags.</param>
+        /// <param name="scope">The optional Meter scope.</param>
+        /// <remarks>
+        /// You can use the scope object to link several Meters with a particular scope.
+        /// For instance, a dependency injection container can choose to associate all Meters that are created within the container with its own scope.
+        /// If the scope object is null, it indicates that the Meter is not linked to any particular scope.
+        /// </remarks>
+        public Meter(string name, string? version, IEnumerable<KeyValuePair<string, object?>>? tags, object? scope = null)
+        {
+            Initialize(name, version, tags, scope);
+            Debug.Assert(Name is not null);
+        }
+
+        private void Initialize(string name, string? version, IEnumerable<KeyValuePair<string, object?>>? tags, object? scope = null)
         {
             Name = name ?? throw new ArgumentNullException(nameof(name));
             Version = version;
+            if (tags is not null)
+            {
+                Tags = new List<KeyValuePair<string, object?>>(tags);
```

Comment:
Any reason to use List rather than array? We should never modify this collection in the future. Also I'd suggest we sort it.

Reply:
I changed the List to Array. I'll address the tags sorting and tags insensitive comparisons in a separate PR when addressing https://github.com/dotnet/runtime/issues/86614.